chia/consensus/block_creation.py

```diff
@@ -213,8 +214,15 @@ def create_foliage(
         additions_root = addition_merkle_set.get_root()
         removals_root = removal_merkle_set.get_root()
 
-        generator_hash = solution_program.get_tree_hash() if solution_program is not None else bytes32([0] * 32)
-        generator_refs_hash = bytes32([1] * 32)
+        generator_hash = bytes32([0] * 32)
+        if solution_program is not None:
+            generator_hash = std_hash(solution_program)
+
+        generator_refs_hash = bytes32([0] * 32)
```

Comment:
@mariano54 I'm pretty sure this will cause a hard fork if blocks have been created with `bytes32([1] * 32)`, and we also enforce bytes32([0] * 32) for blocks with no gen_ref_list